NEWS.md

```diff
@@ -3,6 +3,10 @@ Julia v1.11 Release Notes
 
 New language features
 ---------------------
+* `public` is a new keyword! It behaves analogously to `export` but marks symbols as public
+  without exporting into the namespace of folks who call `using`. Please mark your public but
+  non-exported symbols as `public`. To mark public symbols while keeping compatibility with
+  older versions of Julia, use `@public` from Compat.jl. ([#50105])
 
```

Comment:
It's a bit strange to compare it to `export` since the only thing `export` (make symbols using after `using`) is not something `public` does. Perhaps something like this would be more clear:
```suggestion
* `public` is a new keyword. Symbols marked with `public` are considered public
  API. Symbols marked with `export` are now also treated as public API. The
  difference between `public` and `export` is that `public` names do not become
  available when `using` a package/module. ([#50105])
```